File: app/src/main/java/org/tvheadend/tvhclient/data/entity/ChannelTag.java
package org.tvheadend.tvhclient.data.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;

import java.util.List;

@Entity(tableName = "channel_tags", primaryKeys = {"id", "connection_id"})
public class ChannelTag {

    @ColumnInfo(name = "id")
    private int tagId;                      // u32   required   ID of tag.
    @ColumnInfo(name = "tag_name")
    private String tagName;                 // str   required   Name of tag.
    @ColumnInfo(name = "tag_index")
    private int tagIndex;                   // u32   optional   Index value for sorting (default by from min to max) (Added in version 18).
    @ColumnInfo(name = "tag_icon")
    private String tagIcon;                 // str   optional   URL to an icon representative for the channel.
    @ColumnInfo(name = "tag_titled_icon")
    private int tagTitledIcon;              // u32   optional   Icon includes a title
    @ColumnInfo(name = "channel_count")
    private int channelCount;

    @ColumnInfo(name = "connection_id")
    private int connectionId;
    @ColumnInfo(name = "is_selected")
    private int isSelected;

    @Ignore
    private List<Integer> members;          // u32[] optional   Channel IDs of those that belong to the tag

    public int getTagId() {
        return tagId;
    }

    public void setTagId(int tagId) {
        this.tagId = tagId;
    }

    public String getTagName() {
        return tagName;
    }

    public void setTagName(String tagName) {
        this.tagName = tagName;
    }

    public int getTagIndex() {
        return tagIndex;
    }

    public void setTagIndex(int tagIndex) {
        this.tagIndex = tagIndex;
    }

    public String getTagIcon() {
        return tagIcon;
    }

    public void setTagIcon(String tagIcon) {
        this.tagIcon = tagIcon;
    }

    public int getTagTitledIcon() {
        return tagTitledIcon;
    }

    public void setTagTitledIcon(int tagTitledIcon) {
        this.tagTitledIcon = tagTitledIcon;
    }

    public List<Integer> getMembers() {
        return members;
    }

    public void setMembers(List<Integer> members) {
        this.members = members;
    }

    public int getChannelCount() {
        return channelCount;
    }

    public void setChannelCount(int channelCount) {
        this.channelCount = channelCount;
    }

    public int getConnectionId() {
        return connectionId;
    }

    public void setConnectionId(int connectionId) {
        this.connectionId = connectionId;
    }

    public int getIsSelected() {
        return isSelected;
    }

    public void setIsSelected(int isSelected) {
        this.isSelected = isSelected;
    }
}

File: data/src/main/java/org/tvheadend/data/entity/ChannelTag.kt
package org.tvheadend.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore

data class ChannelTag(

        var tagId: Int = 0,                     // u32   required   ID of tag.
        var tagName: String? = "",               // str   required   Name of tag.
        var tagIndex: Int = 0,                  // u32   optional   Index value for sorting (default by from min to max) (Added in version 18).
        var tagIcon: String? = null,            // str   optional   URL to an icon representative for the channel.
        var tagTitledIcon: Int = 0,             // u32   optional   Icon includes a title
        var channelCount: Int = 0,
        var connectionId: Int = 0,
        var isSelected: Boolean = false,

        var members: List<Int>? = null          // u32[] optional   Channel IDs of those that belong to the tag
)

@Entity(tableName = "channel_tags", primaryKeys = ["id", "connection_id"])
internal data class ChannelTagEntity(

        @ColumnInfo(name = "id")
        var tagId: Int = 0,                     // u32   required   ID of tag.
        @ColumnInfo(name = "tag_name")
        var tagName: String? = "",               // str   required   Name of tag.
        @ColumnInfo(name = "tag_index")
        var tagIndex: Int = 0,                  // u32   optional   Index value for sorting (default by from min to max) (Added in version 18).
        @ColumnInfo(name = "tag_icon")
        var tagIcon: String? = null,            // str   optional   URL to an icon representative for the channel.
        @ColumnInfo(name = "tag_titled_icon")
        var tagTitledIcon: Int = 0,             // u32   optional   Icon includes a title
        @ColumnInfo(name = "channel_count")
        var channelCount: Int = 0,
        @ColumnInfo(name = "connection_id")
        var connectionId: Int = 0,
        @ColumnInfo(name = "is_selected")
        var isSelected: Boolean = false,

        @Ignore
        var members: List<Int>? = null          // u32[] optional   Channel IDs of those that belong to the tag
) {
    companion object {
        fun from(channel: ChannelTag): ChannelTagEntity {
            return ChannelTagEntity(
                    channel.tagId,
                    channel.tagName,
                    channel.tagIndex,
                    channel.tagIcon,
                    channel.tagTitledIcon,
                    channel.channelCount,
                    channel.connectionId,
                    channel.isSelected,
                    channel.members
            )
        }
    }

    fun toChannelTag(): ChannelTag {
        return ChannelTag(tagId, tagName, tagIndex, tagIcon, tagTitledIcon, channelCount, connectionId, isSelected, members)
    }
}
